### PR TITLE
PHPCS: Use JSON instead of plain text

### PR DESCRIPTION
**Pull request by kraftbj**
When trying to edit the list of excluded files, my IDE changed something with the line endings, breaking the scripts.

This PR merges into `phpcs/3rd-party` where it was first discovered.

#### Changes proposed in this Pull Request:
* Converted the list to JSON instead and use that both places.

#### Jetpack product discussion
n/a

#### Does this pull request change what data or activity we track or use?
n/a

#### Testing instructions:
* phpcs -p -s '--filter=bin/class-jetpack-phpcs-exclude-filter.php' '--report=emacs' '--standard=.github/files/php-linting-phpcs.xml'
* Commit something
* Do both the first phpcs command and the pre-commit hook work?

#### Proposed changelog entry for your changes:
* n/a

### CODE REVIEW

File: bin/phpcs-excludelist.json [hunk 1]
```diff
@@ -0,0 +1,604 @@
+[
+    "bin\/build-asset-cdn-json.php",
```
Comment (anomiex) [inline bin/phpcs-excludelist.json:2]: Let's get rid of the extraneous backslashes, and use tabs for indenting. I pushed a464f02e60d944c726a46b078111d4e8ec4f5f13 for that if you want to pull that into this PR.

**Review by anomiex — COMMENTED** [review]
Looks ok, one nitpick.

File: bin/phpcs-excludelist.json (hunk 1 above)
Comment (kraftbj) [reply bin/phpcs-excludelist.json:2]: Cool, will do. I just piped the textlist into PHP and let it json_encode with pretty print do magic. Hold no option on the formatting.

**Comment by anomiex** [comment]
FYI, I submitted #18068 which should hopefully fix your environment being able to commit a line-ending change again. But no objection to also doing this as long as we don't think we'll want comments in the file at any point.

**Comment by anomiex** [comment]
(switching to "Needs Author Reply" pending the import of a464f02 or the like)

**Comment by kraftbj** [comment]
Re comments, understood. If there is a file that really cant be brought into compliance, I'd rather us disable PHPCS (or whatever particular rules) at the top of the file, then keep an exclude list long-term.

Perfect world for me would be never add anything and slowly get this list down to zero within a year or two.

**Pull request merged**